fast_neural_style/net.py

```diff
@@ -0,0 +1,165 @@
+        h = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h = F.relu(self.conv3_1(h))
+        h = F.relu(self.conv3_2(h))
+        h = F.relu(self.conv3_3(h))
+        h = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h = F.relu(self.conv4_1(h))
+        h = F.relu(self.conv4_2(h))
+        h = F.relu(self.conv4_3(h))
+        h = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h = F.relu(self.conv5_1(h))
+        h = F.relu(self.conv5_2(h))
+        h = F.relu(self.conv5_3(h))
+        h = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h = h.view(h.size(0), -1)
+        h = self.dropout1(F.relu(self.fc6(h)))
+        h = self.dropout2(F.relu(self.fc7(h)))
+        h = self.fc8(h)
+        return h
+
+class VGGFeature(nn.Module):
+    def __init__(self):
+        super(VGGFeature, self).__init__()
+        self.conv1_1=nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1)
+        self.conv1_2=nn.Conv2d(64, 64, kernel_size=3, stride=1, padding=1)
+
+        self.conv2_1=nn.Conv2d(64, 128, kernel_size=3, stride=1, padding=1)
+        self.conv2_2=nn.Conv2d(128, 128, kernel_size=3, stride=1, padding=1)
+
+        self.conv3_1=nn.Conv2d(128, 256, kernel_size=3, stride=1, padding=1)
+        self.conv3_2=nn.Conv2d(256, 256, kernel_size=3, stride=1, padding=1)
+        self.conv3_3=nn.Conv2d(256, 256, kernel_size=3, stride=1, padding=1)
+
+        self.conv4_1=nn.Conv2d(256, 512, kernel_size=3, stride=1, padding=1)
+        self.conv4_2=nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+        self.conv4_3=nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+
+        self.conv5_1=nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+        self.conv5_2=nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+        self.conv5_3=nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+
+    def forward(self, x):
+        h  = F.relu(self.conv1_1(x))
+        h  = F.relu(self.conv1_2(h))
+        f1 = h
+        h  = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h  = F.relu(self.conv2_1(h))
+        h  = F.relu(self.conv2_2(h))
+        f2 = h
+        h  = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h  = F.relu(self.conv3_1(h))
+        h  = F.relu(self.conv3_2(h))
+        h  = F.relu(self.conv3_3(h))
+        f3 = h
+        h  = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        h  = F.relu(self.conv4_1(h))
+        h  = F.relu(self.conv4_2(h))
+        h  = F.relu(self.conv4_3(h))
+        f4 = h
+        # h  = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        # h  = F.relu(self.conv5_1(h))
+        # h  = F.relu(self.conv5_2(h))
+        # h  = F.relu(self.conv5_3(h))
+        # h  = F.max_pool2d(h, kernel_size=2, stride=2)
+
+        return [f1,f2,f3,f4] #[relu1_2,relu2_2,relu3_3,relu4_3]
+
+class ResidualBlock(nn.Module):
+    def __init__(self,num):
+        super(ResidualBlock, self).__init__()
+        self.c1=nn.Conv2d(num, num, kernel_size=3, stride=1, padding=1)
```

Comment:
I had bad border artifacts when using zero-padding, so I used reflection padding at the very beginning of the network to ensure that the input and output have the same size.